### DISCUSSION TITLE
Error on iOS with CommunityToolkit.Maui.MediaElement controller

### DISCUSSION THREAD

**Discussion by ghost** [CommunityToolkit/Maui#1428] [General]
On MAUI Project in Visual Studio 2022 when I run the project on iOS simulator

on the Program.cs under Platforms/iOS folder on below line

	UIApplication.Main(args, null, typeof(AppDelegate));
it gives below error

ObjCRuntime.ObjCException: 'Objective-C exception thrown. Name: UIViewControllerHierarchyInconsistency Reason: child view controller:<AVPlayerViewController: 0x14cdaa200> should have parent view controller:<Microsoft_Maui_Controls_Handlers_Items_CarouselViewController: 0x14bd9aa90> but actual parent is:<Microsoft_Maui_Controls_Platform_Compatibility_ShellFlyoutRenderer: 0x14bd55530>

My XAML PAge is like below

<ContentPage.Resources>
    <ResourceDictionary>
        <converters:NotNullOrEmptyStringConverter x:Key="NotNullOrEmptyStringConverter"/>
    </ResourceDictionary>
</ContentPage.Resources>

<ContentPage.BindingContext>
    <local:CardVM  />
</ContentPage.BindingContext>

<CarouselView x:Name="carouselView" 
          ItemsSource="{Binding Cards}" 
          RemainingItemsThreshold="2" 
          RemainingItemsThresholdReached="OnRemainingItemsThresholdReached" 
          VerticalOptions="FillAndExpand" 
          HorizontalOptions="FillAndExpand"
              ItemsLayout="VerticalList" >
    <CarouselView.ItemTemplate>
        <DataTemplate>
            <ScrollView>
                <StackLayout>

                    
                    <Label Text="{Binding Text}" 
                       FontSize="Large" 
                       HorizontalOptions="CenterAndExpand" 
                       VerticalOptions="CenterAndExpand"
                       IsVisible="{Binding Text, Converter={StaticResource NotNullOrEmptyStringConverter}}"/>

                    

                    <Image  
                               Source="{Binding ImageSource}" 
                               HorizontalOptions="FillAndExpand"
                               VerticalOptions="FillAndExpand"
                               Aspect="AspectFill"
                               IsVisible="{Binding ImageSource, Converter={StaticResource NotNullOrEmptyStringConverter}}" />

                    <toolkit:MediaElement Source="https://jlmscloudstorelocal.blob.core.windows.net/audiofiles/video1.mp4" 
                                          ShouldAutoPlay="True"
                                          ShouldKeepScreenOn="True"
                                          ShouldLoopPlayback="False"
                                          ShouldShowPlaybackControls="True"
                          IsVisible="{Binding VideoSource, Converter={StaticResource NotNullOrEmptyStringConverter}}" 
                          HorizontalOptions="FillAndExpand" 
                          VerticalOptions="FillAndExpand"
                                          Aspect="AspectFill"
                                          />

                </StackLayout>
            </ScrollView>
        </DataTemplate>
    </CarouselView.ItemTemplate>
</CarouselView>
I have tried with latest 2.0 version of CommunityToolkit.Maui.MediaElement as well as with 1.0.2 version. It gives error in both version.

Can anyone help with this?

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
